Merge baselin branch into master

FlorianReiss opened the pull request:
* extract only subset of data
* provide code for baseline solution

FlorianReiss committed:
add baseline code

FlorianReiss committed:
add build instructions to README

FlorianReiss committed:
delete old json files

FlorianReiss committed:
add physics description

FlorianReiss committed:
only extarct subset of data

FlorianReiss committed:
Merge remote-tracking branch 'origin/baseline' into baseline

FlorianReiss committed:
update extraction README

particleist commented:
Could you please add the scoring criterion which you implemented into the overleaf and update rapid-2018-vertexing.pdf ?

particleist committed:
Update README.md

Added explicit instructions to copy the libraries into the submissions/starting_kit subdirectory

particleist commented:
I tested this recompiling from scratch and copying the libraries from the build directory and get this segfault again, could you please check/fix?

`Traceback (most recent call last):
  File "/Users/gligorov/anaconda2/envs/vertex_finding/bin/ramp_test_submission", line 11, in <module>
    sys.exit(ramp_test_submission())
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/command_line.py", line 93, in ramp_test_submission
    retrain=retrain)
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/testing.py", line 114, in assert_submission
    fold, ramp_data_dir)
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/submission.py", line 190, in run_submission_on_cv_fold
    ('test', predictions_test)]),
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/scoring.py", line 91, in score_matrix
    for score_type in score_types] for step in ground_truth]
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/scoring.py", line 91, in <listcomp>
    for score_type in score_types] for step in ground_truth]
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/utils/scoring.py", line 91, in <listcomp>
    for score_type in score_types] for step in ground_truth]
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/rampwf/score_types/base.py", line 21, in score_function
    return self.__call__(y_true, y_pred)
  File "./problem.py", line 39, in __call__
    checker.load_from_ramp(y_true_label_index, y_pred_label_index)
  File "./problem.py", line 117, in load_from_ramp
    self.load_data(RecPV_arr_tot, MCPV_arr_tot)
  File "./problem.py", line 95, in load_data
    self.df_rec_pvs.columns=['x', 'y', 'z','matched', 'matched_pv_key']
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/pandas/core/generic.py", line 4389, in __setattr__
    return object.__setattr__(self, name, value)
  File "pandas/_libs/properties.pyx", line 69, in pandas._libs.properties.AxisProperty.__set__
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/pandas/core/generic.py", line 646, in _set_axis
    self._data.set_axis(axis, labels)
  File "/Users/gligorov/anaconda2/envs/vertex_finding/lib/python3.6/site-packages/pandas/core/internals.py", line 3323, in set_axis
    'values have {new} elements'.format(old=old_len, new=new_len))
ValueError: Length mismatch: Expected axis has 3 elements, new values have 5 elements`

FlorianReiss committed:
fix crash when event has no reconstructed pvs

FlorianReiss committed:
remove data frames holding all events in PVChecker

FlorianReiss committed:
only load data of efficiency checker once

FlorianReiss committed:
remove std::cout in PV fitting

FlorianReiss committed:
Merge remote-tracking branch 'origin/baseline' into baseline

FlorianReiss committed:
update scoring description
